### ISSUE TITLE
BUG: AttributeError: 'Handler' object has no attribute '_run'

### ISSUE THREAD

**Issue by DhruvilKarani** [neptune-ai/kedro-neptune#32]
### Describe the bug

The Neptune-Kedro integration is unable to initialize. Raises the above error in __init__.py [line 374]

### Reproduction

Followed the installation mentioned [here](https://pypi.org/project/kedro-neptune/). 

Complete requirements.txt used

```
black==21.5b1
flake8>=3.7.9, <4.0
ipython~=7.10
isort~=5.0
jupyter~=1.0
jupyter_client>=5.1, <7.0
jupyterlab~=3.0
kedro==0.17.7
kedro-viz
kedro-telemetry~=0.1.0
nbstripout~=0.4
pytest-cov~=3.0
pytest-mock>=1.7.1, <2.0
pytest~=6.2
wheel>=0.35, <0.37
scikit-learn
numpy
pandas
tqdm
kaggle
optuna
scikit-plot
neptune-client 
kedro-neptune
neptune-optuna

```

### Traceback

```
Traceback (most recent call last):
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/bin/kedro", line 8, in <module>
    sys.exit(main())
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/kedro/framework/cli/cli.py", line 246, in main
    cli_collection()
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/kedro/framework/cli/cli.py", line 189, in main
    **extra,
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/kedro/framework/cli/project.py", line 399, in run
    pipeline_name=pipeline,
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/kedro/framework/session/session.py", line 387, in run
    run_params=record_data, pipeline=filtered_pipeline, catalog=catalog
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/kedro_neptune/__init__.py", line 447, in before_pipeline_run
    log_data_catalog_metadata(namespace=current_namespace, catalog=catalog)
  File "/Users/dhruvilkarani/opt/anaconda3/envs/blog/lib/python3.7/site-packages/kedro_neptune/__init__.py", line 374, in log_data_catalog_metadata
    if dataset.exists() and not namespace._run.exists(join_paths(namespace._path, name)):
AttributeError: 'Handler' object has no attribute '_run'
```

### Environment
**The output of `pip list`:**
```
Package                Version
---------------------- ---------
aiofiles               0.6.0
alembic                1.7.7
anyconfig              0.10.1
anyio                  3.5.0
appdirs                1.4.4
appnope                0.1.2
argon2-cffi            21.3.0
argon2-cffi-bindings   21.2.0
arrow                  1.2.2
attrs                  21.4.0
autopage               0.5.0
Babel                  2.9.1
backcall               0.2.0
beautifulsoup4         4.10.0
binaryornot            0.4.4
black                  21.5b1
bleach                 4.1.0
boto3                  1.21.19
botocore               1.24.19
bravado                11.0.3
bravado-core           5.17.0
cached-property        1.5.2
cachetools             4.2.4
certifi                2021.10.8
cffi                   1.15.0
chardet                4.0.0
click                  7.1.2
cliff                  3.10.1
cmaes                  0.8.2
cmd2                   2.4.0
colorlog               6.6.0
cookiecutter           1.7.3
coverage               6.3.2
cycler                 0.11.0
debugpy                1.5.1
decorator              5.1.1
defusedxml             0.7.1
dynaconf               3.1.7
entrypoints            0.4
fastapi                0.66.1
flake8                 3.9.2
fonttools              4.30.0
fqdn                   1.5.1
fsspec                 2022.1.0
future                 0.18.2
gitdb                  4.0.9
GitPython              3.1.27
graphql-core           3.1.7
greenlet               1.1.2
h11                    0.13.0
httptools              0.1.2
idna                   2.10
importlib-metadata     4.11.3
importlib-resources    5.4.0
iniconfig              1.1.1
ipykernel              6.9.2
ipython                7.32.0
ipython-genutils       0.2.0
ipywidgets             7.6.5
isoduration            20.11.0
isort                  5.10.1
jedi                   0.18.1
Jinja2                 3.0.3
jinja2-time            0.2.0
jmespath               0.10.0
joblib                 1.1.0
json5                  0.9.6
jsonpointer            2.2
jsonref                0.2
jsonschema             4.4.0
jupyter                1.0.0
jupyter-client         6.1.12
jupyter-console        6.4.2
jupyter-core           4.9.2
jupyter-server         1.15.3
jupyterlab             3.3.2
jupyterlab-pygments    0.1.2
jupyterlab-server      2.10.3
jupyterlab-widgets     1.0.2
kaggle                 1.5.12
kedro                  0.17.7
kedro-neptune          0.0.7
kedro-telemetry        0.1.4
kedro-viz              4.3.1
kiwisolver             1.4.0
Mako                   1.2.0
MarkupSafe             2.1.0
matplotlib             3.5.1
matplotlib-inline      0.1.3
mccabe                 0.6.1
mistune                0.8.4
monotonic              1.6
msgpack                1.0.3
mypy-extensions        0.4.3
nbclassic              0.3.6
nbclient               0.5.13
nbconvert              6.4.4
nbformat               5.2.0
nbstripout             0.5.0
neptune-client         0.15.1
neptune-optuna         0.9.14
nest-asyncio           1.5.4
networkx               2.6.3
notebook               6.4.9
notebook-shim          0.1.0
numpy                  1.21.5
oauthlib               3.2.0
optuna                 2.10.0
packaging              21.3
pandas                 1.3.5
pandocfilters          1.5.0
parso                  0.8.3
pathspec               0.9.0
pbr                    5.8.1
pep517                 0.12.0
pexpect                4.8.0
pickleshare            0.7.5
Pillow                 9.0.1
pip                    22.0.4
pip-tools              6.5.1
plotly                 5.6.0
pluggy                 0.13.1
poyo                   0.5.0
prettytable            3.2.0
prometheus-client      0.13.1
prompt-toolkit         3.0.28
psutil                 5.9.0
ptyprocess             0.7.0
py                     1.11.0
pycodestyle            2.7.0
pycparser              2.21
pydantic               1.9.0
pyflakes               2.3.1
Pygments               2.11.2
PyJWT                  2.3.0
pyparsing              3.0.7
pyperclip              1.8.2
pyrsistent             0.18.1
pytest                 6.2.5
pytest-cov             3.0.0
pytest-mock            1.13.0
python-dateutil        2.8.2
python-dotenv          0.19.2
python-json-logger     2.0.2
python-multipart       0.0.5
python-slugify         6.1.1
pytz                   2021.3
PyYAML                 5.4.1
pyzmq                  22.3.0
qtconsole              5.2.2
QtPy                   2.0.1
regex                  2022.3.2
requests               2.25.1
requests-oauthlib      1.3.1
rfc3339-validator      0.1.4
rfc3987                1.3.8
rope                   0.21.1
ruamel.yaml            0.17.10
ruamel.yaml.clib       0.2.6
s3transfer             0.5.2
scikit-learn           1.0.2
scikit-plot            0.3.7
scipy                  1.7.3
semver                 2.13.0
Send2Trash             1.8.0
sentinel               0.3.0
setuptools             60.9.3
simplejson             3.17.6
six                    1.16.0
sklearn                0.0
smmap                  5.0.0
sniffio                1.2.0
soupsieve              2.3.1
SQLAlchemy             1.4.32
starlette              0.14.2
stevedore              3.5.0
strawberry-graphql     0.93.23
swagger-spec-validator 2.7.4
tenacity               8.0.1
terminado              0.13.3
testpath               0.6.0
text-unidecode         1.3
threadpoolctl          3.1.0
toml                   0.10.2
tomli                  2.0.1
toposort               1.7
tornado                6.1
tqdm                   4.63.0
traitlets              5.1.1
typed-ast              1.5.2
typing_extensions      4.1.1
uri-template           1.2.0
urllib3                1.26.8
uvicorn                0.13.4
uvloop                 0.16.0
watchgod               0.8
wcwidth                0.2.5
webcolors              1.11.1
webencodings           0.5.1
websocket-client       1.3.1
websockets             8.1
wheel                  0.36.2
widgetsnbextension     3.5.2
zipp                   3.7.0
```
OS - macOS 12.0.1 Monterey

Python Version - 3.7.1

### Additional context
The problem does not seem to be in the application code, but in the initialization of neptune packages when running the kedro run. However, here is the nodes.py

```
"""
This is a boilerplate pipeline 'data_science'
generated using Kedro 0.17.7
"""
from distutils.log import Log
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import cross_val_score
from sklearn.metrics import accuracy_score
from kedro.io import MemoryDataSet
import pandas as pd
import numpy as np
import optuna
import neptune.new as neptune
import neptune.new.integrations.optuna as optuna_utils
import matplotlib.pyplot as plt
from scikitplot.metrics import plot_confusion_matrix


def train_model(df_train, y_label, features, neptune_run):
    X_train = df_train[features].values
    y_train = df_train[y_label].values

    def objective(trial):
        n_estimators = trial.suggest_int('n_estimators', 2, 50)
        max_depth = int(trial.suggest_loguniform('max_depth', 1, 20))
        model = RandomForestClassifier(n_estimators=n_estimators, max_depth=max_depth)
        return cross_val_score(model, X_train, y_train, 
            n_jobs=-1, cv=5).mean()

    study = optuna.create_study(direction='maximize')
    neptune_callback = optuna_utils.NeptuneCallback(neptune_run)
    study.optimize(objective, n_trials=100, callbacks=[neptune_callback])
    model = RandomForestClassifier(**study.best_params)
    model.fit(X_train, y_train)

    y_pred = model.predict(X_train)
    acc = accuracy_score(y_train, y_pred)
    neptune_run['nodes/report/train_accuracy'] = acc
    fig, ax = plt.subplots()
    plot_confusion_matrix(y_train, y_pred, ax=ax)
    neptune_run['nodes/report/train_confusion_matrix'].upload(fig)
    return model, {"acc":acc}, {"features":features, "model_type": "RandomForest"}

def evaluate_model(model, df_test, y_label, features, neptune_run):
    X_test = df_test[features].values
    y_test = df_test[y_label].values
    y_pred = model.predict(X_test)
    acc = accuracy_score(y_test, y_pred)
    neptune_run['nodes/report/test_accuracy'] = acc
    fig, ax = plt.subplots()
    plot_confusion_matrix(y_test, y_pred, ax=ax)
    neptune_run['nodes/report/test_confusion_matrix'].upload(fig)
    return {"acc": accuracy_score(y_test, y_pred)}

```

pipeline.py

```
"""
This is a boilerplate pipeline 'data_science'
generated using Kedro 0.17.7
"""

from kedro.pipeline import Pipeline, node, pipeline
from tutorial.pipelines.data_science.nodes import train_model, evaluate_model


def create_pipeline(**kwargs) -> Pipeline:
    return pipeline([
        node(
            func=train_model,
            inputs=["train", "params:y_label", "params:features", "neptune_run"],
            outputs=["model", "train_metrics", "features"],
            name="train_model"
        ),
        node(
            func=evaluate_model,
            inputs=["model", "test", "params:y_label", "params:features", "neptune_run"],
            outputs="test_metrics",
            name="evaluate_model"
        )
    ])

```

**Comment by kamil-kaczmarek** [comment]
hey @DhruvilKarani 

thanks for reporting this.

We are now working on it, you can expect fix to be released this week :slightly_smiling_face:

**Comment by kamil-kaczmarek** [comment]
hey @DhruvilKarani ,

Fixed in #31 

Please upgrade to the latest version `0.0.8`.

Thanks again for reporting that :+1:
